Fixes CVE-2021-25290.

Upstream commit:
https://github.com/python-pillow/Pillow/commit/86f02f7c70862a0954bfe8133736d352db978eaa

--- Pillow-5.1.0/src/libImaging/TiffDecode.c
+++ Pillow-5.1.0/src/libImaging/TiffDecode.c
@@ -36,6 +36,10 @@ tsize_t _tiffReadProc(thandle_t hdata, t
 	TRACE(("_tiffReadProc: %d \n", (int)size));
 	dump_state(state);
 
+	if (state->loc > state->eof) {
+		TIFFError("_tiffReadProc", "Invalid Read at loc %d, eof: %d", state->loc, state->eof);
+		return 0;
+	}
 	to_read = min(size, min(state->size, (tsize_t)state->eof) - (tsize_t)state->loc);
 	TRACE(("to_read: %d\n", (int)to_read));
 
